source/blog/2017-01-11-writing-rpm-macros-for-openstack.html.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Writing rpm-macros for OpenStack
+author: chandankumar
+date: 2017-01-11 14:00:00 IST
+tags: blogs,openstack,rdo,rpm-macros
+---
+
+rpm-macros is a short string, always prefixed by % and generally surrounded by curly brackets ({}) which RPM will convert to a different and usually longer string.
+Some macros can take arguments and some can be quite complex.
+In RHEL, Centos and Fedora, Macros are provided by rpm package and from [redhat-rpm-config](https://apps.fedoraproject.org/packages/redhat-rpm-config).
+In RDO, openstack macros are provided by [openstack-macros](http://cbs.centos.org/koji/packageinfo?packageID=3792) which comes from upstream [rpm-packaging](http://git.openstack.org/cgit/openstack/rpm-packaging/) project.
+You can find list of all macros under /usr/lib/rpm/macros.d/ directory.
+
+To see the list of all available macros on your system:
+
+    $ rpm --showrc
+
+For example: %{_bindir} is a rpm-macro which points to binary directory where executables are usually stored.
+
+To evaluate a rpm macro:
+
+    $ rpm --eval %{_bindir}
+
+%py_build is the commonly used rpm-macro in RDO OpenStack packages which points to *python setup.py build* process.
+
+    $ rpm --eval %py_build
+
+**Motivation behind writing rpm-macro for OpenStack**
```

Comment:
writing rpm-macro for OpenStack -> writing a new RPM macro for OpenStack packages